torch/testing/_internal/common_device_type.py

```diff
@@ -252,27 +252,30 @@
 # then inherit from it for your generic test.
 
 
-def _construct_test_name(test_name, op, device_type, dtype):
-    if op is not None:
-        test_name += "_" + op.name.replace('.', '_')
-        if op.variant_test_name:
-            test_name += "_" + op.variant_test_name
-
-    test_name += "_" + device_type
-
-    if dtype is not None and dtype is not _NO_DTYPES:
-        if isinstance(dtype, (list, tuple)):
-            for d in dtype:
-                test_name += "_" + str(d).split('.')[1]
-        else:
-            test_name += "_" + str(dtype).split('.')[1]
-
-    return test_name
+def _dtype_name(dtype):
+    """ Returns the pretty name of the dtype (e.g. torch.int64 -> int64). """
+    return str(dtype).split('.')[1]
+
+
+def _dtype_test_suffix(dtypes):
+    """ Returns the test suffix for a dtype, sequence of dtypes, or None. """
+    if isinstance(dtypes, list) or isinstance(dtypes, tuple):
+        if len(dtypes) == 0:
+            return ''
+        return '_' + '_'.join((_dtype_name(d) for d in dtypes))
+    elif dtypes:
+        return '_{}'.format(_dtype_name(dtypes))
+    else:
+        return ''
 
 
-# Marker class to signify an absense of dtypes
-class _NO_DTYPES(object):
-    pass
+def _update_param_kwargs(param_kwargs, name, value):
+    """ Adds a kwarg with the specified name and value to the param_kwargs dict. """
+    if isinstance(value, list) or isinstance(value, tuple):
+        # Make name plural
+        param_kwargs['{}s'.format(name)] = value
+    elif value:
```

Comment:
Should there be an else or an assert here?

Reply:
This is intentional - nothing should be added to `param_kwargs` for the None case. Added a comment explaining this